### DISCUSSION TITLE
Is burnup and  depletion calculation done in SCONE?

### DISCUSSION THREAD

**Discussion by emilmammadzada99** [CambridgeNuclear/SCONE#30] [General]
Is burnup and  depletion calculation done in SCONE?

**Comment by ChasingNeutrons** [comment]
Unfortunately not yet. Hopefully soon.

**Reply by tzhang0475** [reply]
Is there a plan to do so? Or maybe it will be good to couple with openmc depletion or with ONIX?

**Reply by ChasingNeutrons** [reply]
There's definitely a plan! It's just getting the time... I think we would prefer to have it closely integrated into SCONE as there are more things that can be done. But it's certainly a high priority of ours.

**Reply by tzhang0475** [reply]
I will be very nice if scone could have a depletion model and a multi-physics interface, with delta-tracking I think it could be a huge advantage against other MC codes. Will SCONE be developed continuously?

**Reply by ChasingNeutrons** [reply]
Agreed, delta tracking opens up some nice possibilities in this realm - though I think OpenMC aren't too far from that either. It certainly will be, we have a few postdocs and students working on it for now.

**Reply by tzhang0475** [reply]
yep, I also undertand openmc are not far from this, but somehow they seems stopped developing in this direction, I ve no idea what happend. As we are facing quite a few embarrassing issues using openMC to do depeletion and multi-physics calculation, we will consider to use SCONE as one of our main code for reactor physics research and teaching.

**Reply by tzhang0475** [reply]
Hi, again a question about burnup. We are working on problem related to pebble bed burnup, a issue is that maybe we have to handle hundreds of depletion zone. Using openMC in this case will be very computationally expensive,  so we are considering SCONE for the criticality calculation. However, in the end we need sth to do the depletion calculation, is there a quick way to couple current SCONE calculation to a depletion code like openMC or ONIX?

**Reply by ChasingNeutrons** [reply]
Hi,
Perhaps others can correct me, though I don't think it should be too hard to couple externally to OpenMC/ONIX. The main things we would need are maybe a slightly different microscopic XS tally - at the moment it is a bit like Serpent detectors, i.e., having to make fake materials to score microscopic reaction rates. They would also have to be modified to allow for more MT numbers, though this should be ok to do. There may be some difficulties depending on how sophisticated you would like to be in treating power normalisation and tallying power production, but that should also be a fairly minor addition (or could be fudged with a fixed 200MeV per fission).
Perhaps more involved than this for your purposes would be the geometry itself. We can certainly do many spheres nested inside each other in arbitrary locations, but, for efficiency and robustness, it may be preferable to make a new universe type - maybe! I must say I haven't though too much about which optimizations we could use to help tracking or input preparation for this geometry type.
Given all of that, I can imagine it being ok to use a script to edit compositions between steps and pass XS and flux amplitude information to an external Bateman solver.

**Reply by tzhang0475** [reply]
Hi, thanks for your reply, very helpful! I assume all detectors required for XS generation is already there in SCONE, is it? Regarding geometry, we plan to use DEM to generate pebble position, I think it should not be a big issue at the moment for a primary geometry, but not yet sure how to deal with the pebble flow yet.
